core/src/main/java/org/apache/gravitino/storage/relational/po/JobPO.java

```diff
@@ -0,0 +1,143 @@
+import org.apache.gravitino.meta.AuditInfo;
+import org.apache.gravitino.meta.JobEntity;
+
+@Getter
+@Accessors(fluent = true)
+@EqualsAndHashCode
+@ToString
+public class JobPO {
+
+  private Long jobRunId;
+  private String jobTemplateName;
+  private Long metalakeId;
+  private String jobExecutionId;
+  private String jobRunStatus;
+  private Long jobFinishedAt;
+  private String auditInfo;
+  private Long currentVersion;
+  private Long lastVersion;
+  private Long deletedAt;
+
+  public JobPO() {
+    // Default constructor for JPA
+  }
+
+  @lombok.Builder(setterPrefix = "with")
+  private JobPO(
+      Long jobRunId,
+      String jobTemplateName,
+      Long metalakeId,
+      String jobExecutionId,
+      String jobRunStatus,
+      Long jobFinishedAt,
+      String auditInfo,
+      Long currentVersion,
+      Long lastVersion,
+      Long deletedAt) {
+    Preconditions.checkArgument(jobRunId != null, "jobRunId cannot be null");
+    Preconditions.checkArgument(
+        StringUtils.isNotBlank(jobTemplateName), "jobTemplateName cannot be blank");
+    Preconditions.checkArgument(metalakeId != null, "metalakeId cannot be null");
+    Preconditions.checkArgument(
+        StringUtils.isNotBlank(jobExecutionId), "jobExecutionId cannot be blank");
+    Preconditions.checkArgument(
+        StringUtils.isNotBlank(jobRunStatus), "jobRunStatus cannot be blank");
+    Preconditions.checkArgument(jobFinishedAt != null, "jobFinishedAt cannot be null");
+    Preconditions.checkArgument(StringUtils.isNotBlank(auditInfo), "auditInfo cannot be blank");
+    Preconditions.checkArgument(currentVersion != null, "currentVersion cannot be null");
+    Preconditions.checkArgument(lastVersion != null, "lastVersion cannot be null");
+    Preconditions.checkArgument(deletedAt != null, "deletedAt cannot be null");
+
+    this.jobRunId = jobRunId;
+    this.jobTemplateName = jobTemplateName;
+    this.metalakeId = metalakeId;
+    this.jobExecutionId = jobExecutionId;
+    this.jobRunStatus = jobRunStatus;
+    this.jobFinishedAt = jobFinishedAt;
+    this.auditInfo = auditInfo;
+    this.currentVersion = currentVersion;
+    this.lastVersion = lastVersion;
+    this.deletedAt = deletedAt;
+  }
+
+  public static class JobPOBuilder {
+    // Builder class for JobPO
+    // Lombok will generate the builder methods based on the fields defined in JobPO
+  }
+
+  public static JobPO initializeJobPO(JobEntity jobEntity, JobPOBuilder builder) {
+    // We should not keep the terminated job entities in the database forever, so we set the
+    // current time as the finished timestamp if the job is in a terminal state,
+    // So the entity GC cleaner will clean it up later.
+    long finished = DEFAULT_DELETED_AT;
+    if (jobEntity.status() == JobHandle.Status.CANCELLED
+        || jobEntity.status() == JobHandle.Status.FAILED
+        || jobEntity.status() == JobHandle.Status.SUCCEEDED) {
+      finished = System.currentTimeMillis();
```

Comment:
So the finish time of a `CANCELLING` job is `System.currentTimeMillis()`?  Why do we set it explicitly in `jobEntity `?

Reply:
We will clean the completed job periodically using this field. If not, the completed job will never be cleaned.